scripts/lib-carousel.js

```diff
@@ -187,14 +187,18 @@ function handleGesture() {
 /**
  * Prevent scrolling when swiping for touch devices
  * @param e event
+ * @returns {boolean} true if scrolling is prevented
  */
-function disableScrolling(e) {
-  const isLink = e.target.tagName.toLowerCase() === 'a';
-  const isButton = e.target.tagName.toLowerCase() === 'button';
+function shouldPreventDefaultScrolling(e) {
```

Comment:
You should prefer guard clauses over indented if-statements for readability and stability. for example:

```
function shouldPreventDefaultScrolling(e) {
    // If target is a link or button, allow the default behavior
    if (['a', 'button'].includes(e.target.tagName.toLowerCase())) return false;

    const isMoveY = Math.abs(touchRelY) > Math.abs(touchRelX);

    // If movement is not primarily vertical, allow the default behavior
    if (!isMoveY) return false;

    e.preventDefault();
    return true;
}
```